socket/launchd_enabled.go

```diff
@@ -0,0 +1,53 @@
+//go:build darwin
+package socket
+
+/*
+#include <stdlib.h>
+#include <launch.h>
+*/
+import "C"
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"unsafe"
+)
+
+func LaunchdSockets(address string) ([]int, error) {
+	c_name := C.CString(address)
+	var c_fds *C.int
+	c_cnt := C.size_t(0)
+
+	err := C.launch_activate_socket(c_name, &c_fds, &c_cnt)
+	if err != 0 {
+		return nil, fmt.Errorf("couldn't activate launchd socket: %v", err)
+	}
+
+	length := int(c_cnt)
+	if length < 2 {
+		return nil, fmt.Errorf("expected at least two sockets to be configured in launchd for '%s', found %d", address, length)
+	}
```

Comment:
This seems like it's made to match with your plist. I think a way of making this configurable would be required.  
Maybe expect one socket per `launchd:` in the port config.

Also I don't understand why you create 2 sockets. Is it so one is TCP-ish and the other UDP-ish?  
I don't remember this too much but from a quick look at the code I don't see how those two would actually handle traffic differently.  
If there is a difference, then the `launchd:` in the port config should also be able to specify whether it should be TCP-like or UDP-like.